components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParAuthServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.par.core;
+
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.par.cache.CacheBackedParDAO;
+import org.wso2.carbon.identity.oauth.par.common.ParConstants;
+import org.wso2.carbon.identity.oauth.par.dao.ParDAOFactory;
+import org.wso2.carbon.identity.oauth.par.dao.ParMgtDAO;
+import org.wso2.carbon.identity.oauth.par.exceptions.ParCoreException;
+import org.wso2.carbon.identity.oauth.par.model.ParAuthResponseData;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * Provides authentication services.
+ */
+public class ParAuthServiceImpl implements ParAuthService {
+
+    @Override
+    public ParAuthResponseData generateParAuthResponse(HttpServletResponse response, HttpServletRequest request) {
+
+        String uuid = generateParReqUriUUID();
+        long expiry = ParConstants.EXPIRES_IN_DEFAULT_VALUE_IN_SEC;
+
+        ParAuthResponseData parAuthResponse = new ParAuthResponseData();
+        parAuthResponse.setUuid(uuid);
+        parAuthResponse.setExpiryTime(expiry);
+
+        return parAuthResponse;
+    }
+
+    public void persistParRequest(String uuid, HashMap<String, String> params, long scheduledExpiryTime)
+            throws ParCoreException {
+
+        getCacheBackedParDAO().persistParRequest(uuid, params.get(OAuthConstants.OAuth20Params.CLIENT_ID),
+                scheduledExpiryTime, params);
+    }
+
+    public static CacheBackedParDAO getCacheBackedParDAO() {
+
+        return new CacheBackedParDAO();
```

Comment:
Why we are creating a new cacheBackedDAO each time 

what is the usage of ParDAOFactory ?